pkg/controller/common/certificates/secret.go

```diff
@@ -63,29 +66,78 @@ func HTTPCertSecretVolume(namer name.Namer, name string) volume.SecretVolume {
 	)
 }
 
-type CertificatesSecret v1.Secret
+type CertificatesSecret struct {
+	v1.Secret
+	ca *CA
+}
+
+func NewCertificatesSecret(secret v1.Secret) (*CertificatesSecret, error) {
+	result := CertificatesSecret{Secret: secret}
+	if err := result.parse(); err != nil {
+		return nil, err
+	}
+	return &result, nil
+}
 
 // CAPem returns the certificate of the certificate authority.
-func (s CertificatesSecret) CAPem() []byte {
+func (s *CertificatesSecret) CAPem() []byte {
 	return s.Data[CAFileName]
 }
 
+// CA returns a pointer to the in-memory representation of the CA contained in the secret.
+func (s *CertificatesSecret) CA() *CA {
+	return s.ca
+}
+
 // CertChain combines the certificate of the CA and the host certificate.
-func (s CertificatesSecret) CertChain() []byte {
+func (s *CertificatesSecret) CertChain() []byte {
 	return append(s.CertPem(), s.CAPem()...)
 }
 
-func (s CertificatesSecret) CertPem() []byte {
+func (s *CertificatesSecret) CertPem() []byte {
 	return s.Data[CertFileName]
 }
 
-func (s CertificatesSecret) KeyPem() []byte {
+func (s *CertificatesSecret) KeyPem() []byte {
 	return s.Data[KeyFileName]
 }
 
-// Validate checks that mandatory fields are present.
+func (s *CertificatesSecret) HasFullCA() bool {
```

Comment:
nit: at first I was thinking that a "full CA" is a CA with all the intermediate certificates. Should we name this function `HasPrivateKey` to make its purpose more explicit ?